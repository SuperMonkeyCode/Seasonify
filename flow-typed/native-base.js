/* @flow */
declare module 'native-base' {
  declare var exports: any;
}
