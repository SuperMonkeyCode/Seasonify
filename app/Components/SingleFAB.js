/* @flow */
import { Fab, Icon } from 'native-base';
import React from 'react';

export default function SingleFAB({ icon, onPress }: { icon: string, onPress: Function }) {
  return (
    <Fab
      // containerStyle={{ marginRight: 10 }}
      style={{ backgroundColor: '#5067FF' }}
      position="bottomRight"
      onPress={onPress}
    >
      <Icon name={icon} />
    </Fab>
  );
}
