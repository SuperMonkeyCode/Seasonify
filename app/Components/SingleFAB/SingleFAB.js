/* @flow */
import { Fab, Icon } from 'native-base';
import { pure } from 'recompose';
import React from 'react';

type Props = {
  /* parent */
  icon: string,
  onPress: Function,
  /* connect */
  /* HOC */
};

const enhance = pure;

function SingleFAB({ icon, onPress }: Props) {
  return (
    <Fab
      // containerStyle={{ marginRight: 10 }}
      style={{ backgroundColor: '#5067FF' }}
      position="bottomRight"
      onPress={onPress}
    >
      <Icon name={icon} />
    </Fab>
  );
}

export default enhance(SingleFAB);
