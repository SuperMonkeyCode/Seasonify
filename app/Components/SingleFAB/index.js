/* @flow */
import SingleFAB from './SingleFAB';

export default SingleFAB;
