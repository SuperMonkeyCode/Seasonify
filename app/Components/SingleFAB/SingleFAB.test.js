import { Fab } from 'native-base';
import React from 'react';

import SingleFAB from './SingleFAB';

function setup(specificProps = {}) {
  const props = {
    icon: 'ok',
    onPress: jest.fn(),
    ...specificProps,
  };
  const component = shallowDive(<SingleFAB {...props} />, Fab);
  return {
    component,
    props,
    actions: {},
  };
}

describe('Rendering', () => {
  it('should match', () => {
    const { component } = setup();
    expect(component).toMatchSnapshot();
  });
});

describe('Events', () => {
  const { component, props } = setup();
  const fab = component.find(Fab);

  describe('when calling onPress', () => {
    fab.simulate('press');

    it('calls onPress', () => {
      expect(props.onPress).toBeCalled();
    });
  });
});
