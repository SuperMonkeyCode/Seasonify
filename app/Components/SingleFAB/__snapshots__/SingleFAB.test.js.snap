// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering should match 1`] = `
<Styled(Fab)
  onPress={[Function]}
  position="bottomRight"
  style={
    Object {
      "backgroundColor": "#5067FF",
    }
  }
>
  <Styled(Icon)
    name="ok"
  />
</Styled(Fab)>
`;
